Implements kafka configurations & kafka server refs,selectors

haarchri opened the pull request:
Signed-off-by: Christopher Haar <chhaar30@googlemail.com>

### Description of your changes

Implements kafka configurations & kafka server refs,selectors

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

```
NAME                                            READY   SYNCED   EXTERNAL-NAME
configuration.kafka.aws.crossplane.io/example   True    True     arn:aws:kafka:eu-central-1:255932642927:configuration/example/38682711-254f-4269-b88e-727b019676f3-1
```

informations for refs in kafka server:
```
    Broker Node Group Info:
      Client Subnet Refs:
        Name:  sample-subnet1
        Name:  sample-subnet2
      Client Subnets:
        subnet-f2ccb898
        subnet-d8ca7b94
      Instance Type:  kafka.t3.small
      Security Group Refs:
        Name:  example
      Security Groups:
        sg-0858486fe394890bf
      Storage Info:
        Ebs Storage Info:
          Volume Size:  1
    Configuration Info:
      Arn:  arn:aws:kafka:eu-central-1:255932642927:configuration/example/38682711-254f-4269-b88e-727b019676f3-1
      Arn Ref:
        Name:    example
      Revision:  1
```


[contribution process]: https://git.io/fj2m9

AaronME commented:
/size L

AaronME commented on examples/kafka/cluster.yaml:
```diff
@@ -2,21 +2,26 @@
+        - name: sample-subnet1
+        - name: sample-subnet2
+      securityGroupRefs:
+        - name: example
```
I think we can use `sample-cluster-sg` here

haarchri replied:
yes - i will change it with rebase

AaronME reviewed (changes requested):
One minor nit and needs rebase. Otherwise passes testing to create cluster and configuration!

haarchri commented:
rebased, changed to generated resolvers/referencers & tested


```
NAME                                      READY   SYNCED   EXTERNAL-NAME
cluster.kafka.aws.crossplane.io/example   True   True     arn:aws:kafka:us-east-1:255932642927:cluster/example/40d67ea4-6a6d-4c7e-81c2-e377e1be16b5-13

NAME                                                          READY   SYNCED   EXTERNAL-NAME
configuration.kafka.aws.crossplane.io/example-configuration   True    True     arn:aws:kafka:us-east-1:255932642927:configuration/example-configuration/f0aec530-75d6-433b-a19f-82434aa44dac-13
```

AaronME reviewed (approved):
LGTM!
